File: 5_D/src/Haber.java
import java.util.ArrayList;

public class Haber {
    public ArrayList<Iletisim> iletisimListesi;
    public Mail mail;
    public Bildirim bildirim;

    public Haber(){
        iletisimListesi=new ArrayList<>();
        mail=new Mail();
        bildirim=new Bildirim();
        
        iletisimListesi.add(mail);
        iletisimListesi.add(bildirim);
    }

    public void bilgilendir(Kullanici kullanici, String mesaj){
        for (Iletisim iletisim: iletisimListesi){
            iletisim.bilgiGonder(kullanici,mesaj);
        }
    }
}

File: 5_D/python/Haber.py
from Iletisim import Iletisim
from Mail import Mail
from Bildirim import Bildirim

class Haber:
    def __init__(self):
        self.iletisimListesi=[]
        self.mail=Mail()            #ordaki bilgiyi alıp burda kullanıyorum ilişki
        self.bildirim=Bildirim()
        self.iletisimListesi.append(self.mail)
        self.iletisimListesi.append(self.bildirim)      #javada add işlemi

    def bilgilendir(self,kullanici,mesaj):
        for iletisim in self.iletisimListesi:
            iletisim.bilgiGonder(kullanici, mesaj)

     

    